torch/csrc/inductor/aoti_torch/opaque_tensor.cpp

```diff
@@ -0,0 +1,36 @@
+#include <ATen/Config.h>
+#include <torch/csrc/inductor/aoti_torch/opaque_tensor.h>
+
+#if AT_MKLDNN_ENABLED()
+#include <ideep.hpp>
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+#endif
+
+namespace torch {
+namespace aot_inductor {
+
+#if AT_MKLDNN_ENABLED()
+
+at::Tensor mkldnn_tensor_from_data_ptr(
+    void* data_ptr,
+    at::IntArrayRef dims,
+    at::ScalarType dtype,
+    at::Device device) {
+  auto a = ideep::tensor({dims.vec(), ideep::tensor::data_type::s8}, data_ptr);
+  return at::native::new_with_itensor_mkldnn(std::move(a), dtype, device);
+}
```

Comment:
Oh, I don't think this is a right way of serializing/deserializing mkldnn tensors. The MKLDNN tensor contains both the memory buffer (the `data_ptr` points to) and the metadata that describes the layout of the memory buffer. When the model is frozen, the weight tensors are with blocked layout which is described with the metadata. So, only serializing/deserializing the memory buffer cannot restore all the information held in the MKLDNN tensor. oneDNN folks are adding the serialization support for MKLDNN tensors per our request. I would suggest we wait for the support ready before adding the support for frozen models.

Reply:
We are working on a POC and will tag you when the PR is ready. cc @chunyuan-w